pkg/solana/chainwriter/chain_writer.go

```diff
@@ -220,6 +225,76 @@ func (s *SolanaChainWriterService) FilterLookupTableAddresses(
 	return filteredLookupTables
 }
 
+// CreateATAs first checks if a specified location exists, then checks if the accounts derived from the
+// ATALookups in the ChainWriter's configuration exist on-chain and creates them if they do not.
+func CreateATAs(ctx context.Context, args any, lookups []ATALookup, derivedTableMap map[string]map[string][]*solana.AccountMeta, client client.MultiClient, idl string, feePayer solana.PublicKey) ([]solana.Instruction, error) {
+	createATAInstructions := []solana.Instruction{}
+	for _, lookup := range lookups {
+		// Check if location exists
+		if lookup.Location != "" {
+			// TODO refactor GetValuesAtLocation to not return an error if the field doesn't exist
```

Comment:
Do we still need to address this?

Reply:
+1 above, I had originally considered doing it a different way and forgot to removed this TODO